### PR TITLE
feat(client): support 'info' command in JSON mode

### LINKED ISSUE #465: Add the option to call the `info` command in JSON mode as well
Currently the `info` command only works for CLI mode.

### PR DESCRIPTION
**Pull request by mlegner**
Closes #465

### CODE REVIEW

File: crates/walrus-service/src/cli_utils/cli_output.rs [hunk 1]
```diff
@@ -1,46 +1,62 @@
...
 pub trait CliOutput: Serialize {
-    /// Returns the output as a human-readable string for the CLI.
-    fn cli_output(&self) -> String;
+    /// Writes the output to stdout as a human-readable string for the CLI.
+    fn print_cli_output(&self);
 
-    /// Returns the output as a string formatted depending on the output mode.
-    fn output_string(&self, json: bool) -> Result<String> {
+    /// Writes the output to stdout formatted depending on the output mode.
+    fn print_output(&self, json: bool) -> Result<()> {
```
Comment (giac-mysten) [inline crates/walrus-service/src/cli_utils/cli_output.rs:41]: Question: printing directly is an instance of YAGNI?

File: crates/walrus-service/src/client/responses.rs [hunk 2]
```diff
@@ -144,3 +162,137 @@ pub struct BlobStatusOutput {
     /// The blob's status.
     pub status: BlobStatus,
 }
+
+/// The output of the `info` command.
+#[serde_as]
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InfoOutput {
```
Comment (giac-mysten) [inline crates/walrus-service/src/client/responses.rs:172]: Remark: IIUC, the information content of the JSON command is lower than the text output (some fields are missing). I think it's perfect -- only the examples are missing.

Question: should we write this in the CLI help string?

File: crates/walrus-service/src/cli_utils/cli_output.rs (hunk 1 above)
Comment (mlegner) [reply crates/walrus-service/src/cli_utils/cli_output.rs:41]: Essentially three different reasons:
- YAGNI. 😁 
- Potentially more efficient as we don't create strings that we immediately write.
- Allows using formatting on the printed table, which is not possible when converting to a string.

File: crates/walrus-service/src/client/responses.rs (hunk 2 above)
Comment (mlegner) [reply crates/walrus-service/src/client/responses.rs:172]: Yes, I can mention this in the help string. Do you think I should add the examples to the JSON output as well?

Comment (mlegner) [reply crates/walrus-service/src/client/responses.rs:172]: I actually added the example blobs to the `InfoOutput` in c9c3e9e9a2f52651f4d90943d0003f36acf61bda.

Comment (giac-mysten) [reply crates/walrus-service/src/client/responses.rs:172]: Perfect. Looks great!

**Pull request merged**